### PR TITLE
feat: remove generic entry point version specification requirement for better devex

### PR DESCRIPTION
**Pull request by denniswon**
# Pull Request Checklist

- [ ] Did you add new tests and confirm existing tests pass? (`yarn test`)
- [ ] Did you update relevant docs? (docs are found in the `site` folder, and guidelines for updating/adding docs can be found in the [contribution guide](https://github.com/alchemyplatform/aa-sdk/blob/main/CONTRIBUTING.md))
- [ ] Do your commits follow the [Conventional Commits](https://www.conventionalcommits.org/en/v1.0.0/) standard?
- [ ] Does your PR title also follow the [Conventional Commits](https://www.conventionalcommits.org/en/v1.0.0/) standard?
- [ ] If you have a breaking change, is it [correctly reflected in your commit message](https://www.conventionalcommits.org/en/v1.0.0/#examples)? (e.g. `feat!: breaking change`)
- [ ] Did you run lint (`yarn lint:check`) and fix any issues? (`yarn lint:write`)
- [ ] Did you follow the [contribution guidelines](https://github.com/alchemyplatform/aa-sdk/blob/main/CONTRIBUTING.md)?

---

## PR-Codex overview
This PR updates various TypeScript files across packages to adjust type constraints and improve code readability.

### Detailed summary
- Updated type constraints in multiple TypeScript files for better readability and maintainability
- Added utility method `getDefaultLightAccountFactoryAddress` to retrieve default light account factory address
- Refactored error handling in `InvalidUserOperationError` constructor
- Adjusted type guards and type definitions for better type safety and clarity

> The following files were skipped due to too many changes: `packages/alchemy/src/client/types.ts`, `packages/core/src/types.ts`, `packages/alchemy/src/middleware/gasManager.ts`

> ✨ Ask PR-Codex anything about this PR by commenting with `/codex {your question}`

### CODE REVIEW

**Review by dphilipson — APPROVED** [review]
Nice cleanup!

File: packages/core/src/types.ts [hunk 1]
```diff
@@ -386,12 +387,13 @@ export interface UserOperationStruct_v7 {
...
-    : TEntryPointVersion extends "0.7.0"
-    ? UserOperationStruct_v7
-    : never;
+export type UserOperationStruct<
+  TEntryPointVersion extends EntryPointVersion = EntryPointVersion
+> = TEntryPointVersion extends "0.6.0"
+  ? UserOperationStruct_v6
+  : TEntryPointVersion extends "0.7.0"
+  ? UserOperationStruct_v7
+  : never;
```
Comment (moldy530) [inline packages/core/src/types.ts:396]: we should change this `never` to union type no? 

https://github.com/alchemyplatform/aa-sdk/pull/607/files#diff-887ed5859d7652b1d3596c31911d16babe9c65fe8b94ae1c92762907291791d7R73 would resolve as never no?

File: packages/core/src/types.ts [hunk 2]
```diff
@@ -189,18 +189,19 @@ export interface UserOperationRequest_v7 {
...
-    : TEntryPointVersion extends "0.7.0"
-    ? UserOperationRequest_v7
-    : never;
+export type UserOperationRequest<
+  TEntryPointVersion extends EntryPointVersion = EntryPointVersion
+> = TEntryPointVersion extends "0.6.0"
+  ? UserOperationRequest_v6
+  : TEntryPointVersion extends "0.7.0"
+  ? UserOperationRequest_v7
+  : never;
```
Comment (moldy530) [inline packages/core/src/types.ts:198]: we should change this never to union type no?

https://github.com/alchemyplatform/aa-sdk/pull/607/files#diff-887ed5859d7652b1d3596c31911d16babe9c65fe8b94ae1c92762907291791d7R73 would resolve as never no?

File: packages/core/src/types.ts (hunk 1 above)
Comment (denniswon) [reply packages/core/src/types.ts:396]: hhmm, but it is already a union right? never will never hit.

UoRequest<EntryPointVersion> = v6 | v7

also UoRequest = v6 | v7

File: packages/core/src/types.ts (hunk 2 above)
Comment (denniswon) [reply packages/core/src/types.ts:198]: ditto: as above.

Comment (dphilipson) [reply packages/core/src/types.ts:198]: By the way, I thought I'd point out that another way to write these "type-level switch statements" which is a little shorter is
```ts
export type UserOperationRequest<
  TEntryPointVersion extends EntryPointVersion = EntryPointVersion
> = {
  "0.6.0": UserOperationRequest_v6;
  "0.7.0": UserOperationRequest_v7;
}[TEntryPointVersion];
```
but the current way is fine too.

Comment (moldy530) [reply packages/core/src/types.ts:198]: oh neato @dphilipson ! yea I like that format a lot

**Pull request merged**